parlai/tasks/natural_questions/agents.py

```diff
@@ -0,0 +1,264 @@
+            offset_unit = 'token'
+            context = context.split(' ')
+
+        short_answers = []
+        for annotation in example['annotations']:
+            if 'short_answers' in annotation and annotation['short_answers']:
+                for sa in annotation['short_answers']:
+                    start_ind = sa[f'start_{offset_unit}']
+                    end_ind = sa[f'end_{offset_unit}']
+                    ans = context[start_ind:end_ind]
+                    if self.use_html:
+                        short_answers.append(ans.decode())
+                    else:
+                        short_answers.append(' '.join(ans))
+            elif 'yes_no_answer' in annotation and \
+                    annotation['yes_no_answer'] and \
+                    annotation['yes_no_answer'] != 'NONE':
+                short_answers.append(annotation['yes_no_answer'])
+        return short_answers
+
+    def load_from_chunk(self, chunk_idx: int):
+        """
+        Loads from a chunk of the dataset, given the chunk index.
+
+        Returns a list of dictionaries. Each dictionary is an example from the
+            main dataset and stores the components of that examples (e.g.,
+            contenxt, question, candidate answers etc.) as key-value pairs.
+
+        :param chunk_idx: the index of the chunk dataset chunk file.
+        """
+
+        def _extarct_labels_indices(example, candidate_labels):
+            labels = []
+            for label in example['annotations']:
+                label_ind = label['long_answer']['candidate_index']
+                labels.append(candidate_labels[label_ind])
+            return labels
+
+        fname = f'nq-{self.dtype}-{str(chunk_idx).zfill(2)}.jsonl'
+        fpath = os.path.join(self.dpath, fname)
+        output = []
+        with jsonlines.open(fpath, 'r') as fi:
+            for example in fi:
+                example_components = dict()
+                example = self._simplify(example)
+                context = example[_context_type_key(self.use_html)]
+                question = example['question_text']
+                example_components['text'] = f'{context}\n{question}?'
+
+                if self.use_long_answer:
+                    example_components['long_answers_candidate'] = \
+                        self._get_candidate_long_answers(
+                        example)
+                    example_components['long_answers'] = \
+                        _extarct_labels_indices(example,
+                                                example_components['long_answers_candidate'])
+                else:
+                    example_components['short_answers'] = \
+                        self._get_short_answers(example)
+                output.append(example_components)
+        return output
+
+    def create_message(self, example_components, entry_idx=0):
+        label_key = 'long_answers' if self.use_long_answer else 'short_answers'
+        return {'id': self.id,
```

Comment:
is `long_answers_candidate` a list of possible label candidates? if so, let's add that under the `label_candidates` field